### PR TITLE
Add debounce to fsnotify

### LINKED ISSUE #1809: Re-add debounce to fsnotify
The latest version of fsnotify has removed debouncing from the library, so we likely need to implement this ourselves.

Related to https://github.com/digitallyinduced/ihp/pull/1771

### PR DESCRIPTION
**Pull request by mpscholten**
Fixes #1809

### CODE REVIEW

File: ihp.cabal [hunk 1]
```diff
@@ -105,6 +105,7 @@ common shared-properties
             , with-utf8
             , ihp-hsx
             , ihp-postgresql-simple-extra
+            , auto-update
```
Comment (amitaibu) [inline ihp.cabal:108]: Is this related to the debounce?

Comment (mpscholten) [reply ihp.cabal:108]: Yes the `Control.Debounce` module is part of that package

Comment (amitaibu) [reply ihp.cabal:108]: Should we add a comment to indicate it?

**Review by amitaibu — COMMENTED** [review]
Locally I see the "modules loaded" appears now only twice rather than three times, when I switch git branches.

image

Not sure if we can reduce to a single load?

File: ihp.cabal (hunk 1 above)
Comment (mpscholten) [reply ihp.cabal]: Good point, added 👍

**Comment by mpscholten** [comment]
> Not sure if we can reduce to a single load?

Not possible without adding latency. We could use https://hackage.haskell.org/package/auto-update-0.1.6/docs/Control-Debounce.html#v:trailingEdge to first cool down and then trigger the reloads. The issue is that this adds latency to every normal single file change

**Comment by amitaibu** [comment]
Yeah, I was just reading about the `trailingEdge`, and it seems like a no go.

Thanks 🙏
